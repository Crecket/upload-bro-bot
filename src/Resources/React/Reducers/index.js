import {combineReducers} from "redux";

import modal from "./modalReducers";
import user from "./user";

export default combineReducers({
    modal,
    user,
})
