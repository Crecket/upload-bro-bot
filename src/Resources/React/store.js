import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import promise from "redux-promise-middleware";
import logger from "redux-logger";

// fetch all reducers as a bundle
import reducer from "./Reducers/index.js";

// create middleware with logger
var middleware = applyMiddleware(promise(), thunk)
if (process.env.DEBUG === "true") {
    // create middleware with logger
    middleware = applyMiddleware(promise(), thunk, logger())
}


//return the store
export default createStore(reducer, middleware)
