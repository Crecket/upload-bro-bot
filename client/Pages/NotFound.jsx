import React from 'react';
import {Helmet} from "react-helmet";

class NotFound extends React.Component {
    constructor(props, context) {
        super(props, context);
        this.state = {};
    };

    render() {
        return (
            <div style={{textAlign: 'center', marginTop: 40}}>
                <Helmet>
                    <title>UploadBroBot - 404 Not Found</title>
                </Helmet>
                <h1>Page Not Found</h1>
            </div>
        );
    };
}

export default NotFound;