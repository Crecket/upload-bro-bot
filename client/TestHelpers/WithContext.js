import React from "react";
import PropTypes from "prop-types";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter, StaticRouter } from "react-router-dom";
import { getMuiTheme, muiThemeable } from "material-ui/styles";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

// the theme we want to include in our context
import LightBlue from "../Themes/LightBlue.jsx";

// navigator fallback for server-side rendering
const navigatorHelper =
    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36";

// the muitheme context
const muiTheme = getMuiTheme(LightBlue, { userAgent: navigatorHelper });

// the final context object
const context = { muiTheme: muiTheme };

// render functions with the required context
export const Shallow = (node, options = { location: "/dashboard" }) => {
    return shallow(
        <MemoryRouter initialEntries={[options.location]}>
            <MuiThemeProvider muiTheme={muiTheme}>
                {node}
            </MuiThemeProvider>
        </MemoryRouter>
    );
};

export const Mount = (node, options = { location: "/dashboard" }) => {
    return mount(
        <MemoryRouter initialEntries={[options.location]}>
            <MuiThemeProvider muiTheme={muiTheme}>
                {node}
            </MuiThemeProvider>
        </MemoryRouter>
    );
};

export const Renderer = (node, options = { location: "/dashboard" }) => {
    return renderer.create(
        <MemoryRouter initialEntries={[options.location]}>
            <MuiThemeProvider muiTheme={muiTheme}>
                {node}
            </MuiThemeProvider>
        </MemoryRouter>,
        {
            context: context,
            childContextTypes: { muiTheme: PropTypes.object }
        }
    );
};
