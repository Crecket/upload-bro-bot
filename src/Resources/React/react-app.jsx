import React from 'react';
import ReactDOM  from 'react-dom';
import {Provider} from "react-redux";
import injectTapEventPlugin  from 'react-tap-event-plugin';
import Logger from './Helpers/Logger.js';

// main app
import Main from './Components/Main';
import store from "./store";

// injection, required for materialze tap events
injectTapEventPlugin();

// render the react app
ReactDOM.render(
    <Provider store={store}>
        <Main />
    </Provider>,
    document.getElementById('app')
);

Logger.debug('Mounted react succesfully');
