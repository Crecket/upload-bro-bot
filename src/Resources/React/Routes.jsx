import React from 'react';
import {Route, IndexRoute} from 'react-router';

import Main from './Components/Main';
import Home from './Pages/Home';
import Login from './Pages/Login';
import NotFound from './Pages/NotFound';
import DropboxLoginCallback from './Pages/DropboxLoginCallback';
import GoogleLoginCallback from './Pages/GoogleLoginCallback';

export default (
    <Route path="/" component={Main}>
        {/* default route */}
        <IndexRoute component={Home}/>

        {/* Routes*/}
        <Route path='/login' component={Login}/>
        <Route path='/login/google/callback' component={GoogleLoginCallback}/>
        <Route path='/login/dropbox/callback' component={DropboxLoginCallback}/>

        {/* fall back route, if no others are found show 404*/}
        <Route path='*' component={NotFound}/>
    </Route>
);