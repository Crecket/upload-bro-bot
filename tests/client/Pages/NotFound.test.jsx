"use strict";
import React from "react";

import { Renderer } from "../../WithContext";
import NotFound from "../../../client/Pages/NotFound";

describe("<NotFound />", () => {
    it("matches snapshot", () => {
        const tree = Renderer(<NotFound />);
        expect(tree).toMatchSnapshot();
    });
});
