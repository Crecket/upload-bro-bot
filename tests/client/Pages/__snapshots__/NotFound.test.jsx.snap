// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> matches snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": 40,
      "textAlign": "center",
    }
  }
>
  <h1>
    Page Not Found
  </h1>
</div>
`;
